pinot-common/src/main/java/com/linkedin/pinot/common/metadata/segment/SegmentZKMetadata.java

```
@@ -63,6 +69,16 @@ public SegmentZKMetadata(ZNRecord znRecord) {
     _totalRawDocs = znRecord.getLongField(CommonConstants.Segment.TOTAL_DOCS, -1);
     _crc = znRecord.getLongField(CommonConstants.Segment.CRC, -1);
     _creationTime = znRecord.getLongField(CommonConstants.Segment.CREATION_TIME, -1);
+
+    try {
+      String partitionMetadataJson = znRecord.getSimpleField(CommonConstants.Segment.PARTITION_METADATA);
+      if (partitionMetadataJson != null) {
+        _partitionMetadata = SegmentPartitionMetadata.fromJsonString(partitionMetadataJson);
+      }
+    } catch (IOException e) {
+      LOGGER.error("Exception caught while reading partition info from segment zk metadata.", e);
```

Based on @mcvsubbu 's comments, I decided to swallow the exception and drop partition metadata, but continue with the rest of the flow.